src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -33,27 +35,67 @@ public GenericTypeParameter(string name, IPythonModule declaringModule, IReadOnl
         public override PythonMemberType MemberType => PythonMemberType.Generic;
         public override bool IsSpecialized => true;
 
-
-        public static IPythonType FromTypeVar(IArgumentSet argSet, IPythonModule declaringModule, IndexSpan location = default) {
+        private static bool TypeVarArgumentsValid(IArgumentSet argSet) {
             var args = argSet.Values<IMember>();
+            var eval = argSet.Eval;
+            var callExpression = argSet.CallExpression;
+            var callLocation = callExpression?.GetLocation(eval?.Module);
+
             if (args.Count == 0) {
-                // TODO: report that at least one argument is required.
-                return declaringModule.Interpreter.UnknownType;
+                eval?.ReportDiagnostics(
```

Comment:
I don't think it is possible to have `eval == null` here. It is not possible to run analysis without an evaluator.

Reply:
Ok I have changed it.